src/GGLSetup.py

```diff
@@ -208,13 +256,23 @@ def _install_build_dependencies() -> bool:
             "liburiparser-dev", "cgroup-tools"
         ]
 
-        # Update package list
-        update_command = ["sudo", "apt", "update"]
-        subprocess.run(update_command, check=True)
+        # Check which packages are missing
+        check_cmd = ["dpkg", "-l"] + packages
+        result = subprocess.run(check_cmd, capture_output=True, text=True)
+
+        missing_packages = []
+        for pkg in packages:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

To create a `list`, try to use `list` comprehension instead of a loop. List comprehension is the preferred way to make a list using Python, and it&#39;s simpler and easier to understand than using a loop.

[Learn more](https://docs.python.org/3/tutorial/datastructures.html#list-comprehensions)